src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs

```diff
@@ -664,6 +664,33 @@ public ResultWrapper<IEnumerable<FilterLog>> eth_getLogs(Filter filter)
         {
             return GetFailureResult<IEnumerable<FilterLog>>(exception, _ethSyncingInfo.SyncMode.HaveNotSyncedReceiptsYet());
         }
+
+        ResultWrapper<IEnumerable<FilterLog>> FailWithNoHeadersSyncedYet(SearchResult<BlockHeader> blockResult)
+            => GetFailureResult<IEnumerable<FilterLog>, BlockHeader>(blockResult, _ethSyncingInfo.SyncMode.HaveNotSyncedHeadersYet());
+
+        static bool TryFindOrUseLatest(IBlockFinder blockFinder, ref BlockParameter blockParameter, out SearchResult<BlockHeader> blockResult, out long sourceBlockNumber)
+        {
+            blockResult = blockFinder.SearchForHeader(blockParameter);
+
+            if (blockResult.IsError)
+            {
+                sourceBlockNumber = blockParameter.Type is BlockParameterType.BlockNumber ? blockParameter.BlockNumber.Value : 0;
+
+                if (blockParameter.Type is BlockParameterType.BlockNumber &&
+                    blockFinder.Head?.Number is not null &&
+                    blockFinder.Head.Number < blockParameter.BlockNumber)
```

Comment:
What if `blockFinder.Head.Number == blockParameter.BlockNumber`?

Reply:
It's OK, logs will be returned for a single block